js/classic/ui.js

```diff
@@ -10,42 +10,158 @@
-   *     .onTurnOn -- called when the user tries to turn on the switch.
-   *     .onTurnOff -- called when the user tries to turn off the switch.
-   *     .onCancelTurnOn -- called when the user tries to turn off the switch
-   *         while the current state is "turning on".
-   *     .onCancelTurnOff -- called when the user tries to turn on the switch
-   *         while the current state is "turning off".
-   * @param defaultState the default state of this switch.
-   * @param enable whether this switch is enabled or not in default.
+   * It adds a class "classic-ui" to the DOM element so we can write css
+   * classes only for elements in class-ui's components easily.
+   *
+   * @see _createSelector
    */
-  ui.SwitchButton = function(handler, defaultState, enable) {
-    this._selector = $(window.document.createElement('div')).slider({max: 2});
-    this._selector.addClass('classic-ui-switch-button');
-    this._selector.slider('disable');
+  var _createClassicUISelector = function(nodeName, className) {
+    return _createSelector(nodeName, className).addClass('classic-ui');
+  };
+
+  /**
+   * Base class for ui components.
+   *
+   * An ui component is a bridge for communication between the user and
+   * the backend handler.  It passes all the related events to
+   * specified event handlers gived by the backend handler.  And all
+   * the controlling rights are belone to the backend, which includes
+   * the state, shape, enabling, etc.
+   *
+   * In implementation level, it contains a jquery selector for
+   * managering DOM elements and lots of methods to control it.  While
+   * constructing, the backend should provide a dict of event handlers
+   * to catch various kinds of events trigger by the front end.
+   *
+   * @constructor
+   * @param {jQuery} selector - A jquery selector contains the main DOM
+   *     element.
+   * @param {Object} handler - A dict of event handlers.
+   */
+  ui.Component = function(selector, handler) {
+    this._selector = selector;
     this._handler = handler;
-    this._state = '';
-    this._enable = null;
+  };
 
-    this._selector.click(this._onClick.bind(this));
+  ui.Component.prototype = {
+    /**
+     * @returns {jQuery} Gets the jquery selector which contains the main
+     *     DOM element(s).
+     */
+    get selector() {
+      return this._selector;
+    },
 
-    this.state = (defaultState !== undefined ? defaultState : this.STATE.OFF);
-    this.enable = (enable !== undefined ? enable : true);
+    /**
+     * Destroys this ui component, which includes the jquery selector.
+     *
+     * After calling this method, this ui component is no more useable.
+     */
+    destroy: function() {
+      this.selector.remove();
+    },
+
+    /**
+     * Calls an event handler.
+     *
+     * @param {String} handlerName - Name of the event handler.
+     * @param {=} args - Arguments to pass to the event handler.
```

Comment:
Could you remove the '=' in `{=}` ? jsdoc will failed to parse this tags.
You could either use `{}` or remove `{=}` completely.
That is to say:
```
@param args - Arguments to pass to the event handler.
```
or 
```
@param {} args - Arguments to pass to the event handler.
```